src/Orleans/Serialization/TypeUtilities.cs

```diff
@@ -266,12 +266,7 @@ private static bool AreInternalsVisibleTo(Assembly fromAssembly, Assembly toAsse
             // Check InternalsVisibleTo attributes on the from-assembly, pointing to the to-assembly.
             var serializationAssemblyName = toAssembly.GetName().FullName;
             var internalsVisibleTo = fromAssembly.GetCustomAttributes<InternalsVisibleToAttribute>();
-            if (internalsVisibleTo.All(_ => _.AssemblyName != serializationAssemblyName))
-            {
-                return true;
```

Comment:
Could you have just inverted the condition?
"If there are no assemblies which match the assembly name, then the internals are not visible to this assembly"